libtsuba/src/RDG.cpp

```diff
@@ -389,70 +391,75 @@ tsuba::RDG::DoMake(const katana::Uri& metadata_dir) {
   }
 
   auto edge_result = AddProperties(
-      metadata_dir, core_->part_header().edge_prop_info_list(),
+      metadata_dir, core_->part_header().edge_prop_info_list(), &grp,
       [rdg = this](const std::shared_ptr<arrow::Table>& props) {
         return rdg->core_->AddEdgeProperties(props);
       });
   if (!edge_result) {
     return edge_result.error().WithContext("populating edge properties");
   }
 
+  katana::Uri t_path = metadata_dir.Join(core_->part_header().topology_path());
+  if (auto res = core_->topology_file_storage().Bind(t_path.string(), true);
+      !res) {
+    return res.error();
+  }
+
+  rdg_dir_ = metadata_dir;
+
   const std::vector<PropStorageInfo>& part_prop_info_list =
       core_->part_header().part_prop_info_list();
-  if (!part_prop_info_list.empty()) {
-    auto part_result = AddProperties(
-        metadata_dir, part_prop_info_list,
-        [rdg = this](const std::shared_ptr<arrow::Table>& props) {
-          return rdg->AddPartitionMetadataArray(props);
-        });
-    if (!part_result) {
-      return part_result.error();
-    }
+  if (part_prop_info_list.empty()) {
+    return grp.Finish();
+  }
+
+  auto part_result = AddProperties(
+      metadata_dir, part_prop_info_list, &grp,
+      [rdg = this](const std::shared_ptr<arrow::Table>& props) {
+        return rdg->AddPartitionMetadataArray(props);
+      });
+  if (!part_result) {
+    return part_result.error();
+  }
+  if (auto res = grp.Finish(); !res) {
+    return res.error();
+  }
 
-    if (local_to_user_id_->length() == 0) {
-      // for backward compatibility
-      if (local_to_global_id_->length() !=
-          core_->part_header().metadata().num_nodes_) {
-        return KATANA_ERROR(
-            tsuba::ErrorCode::InvalidArgument,
-            "regenerate partitions: number of Global Node IDs {} does not "
-            "match the number of master nodes {}",
-            local_to_global_id_->length(),
-            core_->part_header().metadata().num_nodes_);
-      }
-      // NB: this is a zero-copy slice, so the underlying data is shared
-      set_local_to_user_id(local_to_global_id_->Slice(0));
-    } else if (
-        local_to_user_id_->length() !=
-        (core_->part_header().metadata().num_owned_ +
-         local_to_global_id_->length())) {
+  if (local_to_user_id_->length() == 0) {
+    // for backward compatibility
+    if (local_to_global_id_->length() !=
+        core_->part_header().metadata().num_nodes_) {
```

Comment:
Do we still need this backward compatibility?

Reply:
Not sure; would prefer not to change the actual functionality here though in case we do.